src/sql/statement.rs

```diff
@@ -709,24 +710,43 @@ fn kafka_sink_builder(
     let encoder = Encoder::new(desc);
     let value_schema = encoder.writer_schema().canonical_form();
 
+    let mut with_options = normalize::with_options(&with_options);
+
+    // Use the user supplied value for replication factor, or default to 3
+    let replication_factor = match with_options.remove("replication_factor") {
+        None => 3,
+        Some(Value::Number(n)) => n.parse::<u32>()?,
+        Some(_) => bail!("replication factor for sink topics has to be a positive integer"),
+    };
+
+    if replication_factor == 0 {
+        bail!("replication factor for sink topics has to be greater than zero");
```

Comment:
Why the different message? ("has to be a positive integer" vs. "has to be greater than zero") ?

Reply:
to be slightly more explicit in the case where they input 0?